core/utils.py

```diff
@@ -0,0 +1,247 @@
+"""
+Utility functions for working with contributor data.
+
+This module provides functions to fetch and parse contributor data from YAML files,
+following the same format used by the Jekyll site and pyosMeta package.
+"""
+
+import yaml
+import logging
+from datetime import datetime
+from typing import List, Dict, Any, Optional
+from urllib.request import urlopen
+from urllib.error import URLError
+
+logger = logging.getLogger(__name__)
+
+
+class ContributorDataError(Exception):
+    """Custom exception for contributor data related errors."""
+    pass
+
+
+def fetch_contributors_yaml(url: str = None) -> List[Dict[str, Any]]:
+    """
+    Fetch contributor data from YAML source.
+    
+    Args:
+        url: URL to fetch YAML from. If None, uses the default pyOpenSci GitHub URL.
+        
+    Returns:
+        List of contributor dictionaries.
+        
+    Raises:
+        ContributorDataError: If data cannot be fetched or parsed.
+    """
+    if url is None:
+        url = "https://raw.githubusercontent.com/pyOpenSci/pyopensci.github.io/main/_data/contributors.yml"
+    
+    try:
+        with urlopen(url) as response:
+            yaml_content = response.read().decode('utf-8')
+            contributors = yaml.safe_load(yaml_content)
+            
+        if not isinstance(contributors, list):
+            raise ContributorDataError("YAML data should be a list of contributors")
+            
+        logger.info(f"Successfully fetched {len(contributors)} contributors from {url}")
+        return contributors
+        
+    except URLError as e:
+        logger.error(f"Failed to fetch contributors from {url}: {e}")
+        raise ContributorDataError(f"Network error: {e}")
+    except yaml.YAMLError as e:
+        logger.error(f"Failed to parse YAML: {e}")
+        raise ContributorDataError(f"YAML parsing error: {e}")
+    except Exception as e:
+        logger.error(f"Unexpected error fetching contributors: {e}")
+        raise ContributorDataError(f"Unexpected error: {e}")
+
+
+def parse_contributor_date(date_str: Any) -> Optional[datetime]:
+    """
+    Parse a date string into a datetime object.
+    
+    Args:
+        date_str: Date string in various formats.
+        
+    Returns:
+        Parsed datetime object or None if parsing fails.
+    """
+    if not date_str:
+        return None
+        
+    # Convert to string if it's not already
+    date_str = str(date_str).strip()
+    
+    # Try common date formats
```

Comment:
I'm curious about this - are we parsing dates rom the yaml file? If we are, then pyosmeta has already cleaned up the date in that file and it should be consistent (but perhaps you saw something that I missed?)